core/arch/aarch64/aarch64.asm

```diff
@@ -115,12 +115,17 @@ GLOBAL_LABEL(unexpected_return:)
         JUMP  GLOBAL_REF(unexpected_return)
         END_FUNC(unexpected_return)
 
-/* All CPU ID registers are accessible only in privileged modes. */
-        DECLARE_FUNC(cpuid_supported)
-GLOBAL_LABEL(cpuid_supported:)
-        mov      w0, #0
+/* bool mrs_id_reg_supported(void)
+ * Checks for support of the MRS instr by attempting to read Instruction Set
+ * Attribute Register 0. Some older kernels on v8.0 systems do not support
+ * this, raising a SIGILL.
```

Comment:
Hmm, I think DR is using `mrs` in multiple other places already?

Reply:
Most of the time the usual status registers are being read: NZCV, FPCR and FPSR.
The other two are CTR_EL0 and TPIDR_EL0 both in user-space at EL0.
Before this patch, this was not a problem, but now we are reading registers at EL1.

That comment is too vague. It should make clear that that the Linux kernel enables MRS to read a certain set of specified registers above exception level EL0 using a trap-and-emulate mechanism. Comment updated.